edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Kubernetes.IntegrationTest/EdgeDeploymentControllerTest.cs

```diff
@@ -0,0 +1,590 @@
+                "config-volume",
+                "configMapName",
+                "trustBundlePath",
+                "trust-bundle-volume",
+                "trustBundleConfigMapName",
+                PortMapServiceType.ClusterIP,
+                persistentVolumeName,
+                storageClassName,
+                Option.Some<uint>(100),
+                "apiVersion",
+                new Uri("http://localhost:35001"),
+                new Uri("http://localhost:35000"),
+                false,
+                false,
+                experimentalFeatures == null ? new Dictionary<string, bool>() : experimentalFeatures);
+            var pvcMapper = new KubernetesPvcMapper(persistentVolumeName, storageClassName, 100);
+            var serviceAccountMapper = new KubernetesServiceAccountMapper();
+            return new EdgeDeploymentController(
+                resourceName,
+                deviceSelector,
+                this.client.DeviceNamespace,
+                this.client.Kubernetes,
+                moduleLifeCycleManager,
+                kubernetesServiceMapper,
+                deploymentMapper,
+                pvcMapper,
+                serviceAccountMapper);
+        }
+
+        private KubernetesModule CreateKubernetesModule(string moduleName, CreatePodParameters createOptions, string imageName)
+        {
+            KubernetesConfig config = new KubernetesConfig("image", createOptions, Option.None<AuthConfig>());
+            IModule m1 = new DockerModule(moduleName, "v1", ModuleStatus.Running, RestartPolicy.Always, new DockerConfig(imageName), ImagePullPolicy.OnCreate, Core.Constants.DefaultPriority, null, null);
+            return new KubernetesModule(m1, config, new KubernetesModuleOwner("v1", "Deployment", "iotedged", "123"));
+        }
+
+        [Fact]
+        public async Task CheckIfCreateDeploymentIsSuccessfulWithNoResources()
+        {
+            var moduleName = "module-a";
+            var deviceSelector = $"{Kubernetes.Constants.K8sEdgeDeviceLabel}=deviceid,{Kubernetes.Constants.K8sEdgeHubNameLabel}=hostname";
+            var moduleLifeCycleManager = new DummyModuleIdentityLifecycleManager(
+                "hostname",
+                "gatewayhostname",
+                "deviceid",
+                moduleName,
+                new ConnectionStringCredentials("connectionString"));
+            var creatOptions = CreatePodParameters.Create();
+            var controller = this.CreateDeploymentController(deviceSelector, moduleLifeCycleManager, string.Empty, string.Empty);
+            string imageName = "test-image:1";
+            KubernetesModule km1 = this.CreateKubernetesModule(moduleName, creatOptions, imageName);
+
+            moduleLifeCycleManager.SetModules(moduleName);
+            await controller.DeployModulesAsync(ModuleSet.Create(km1), ModuleSet.Empty);
+
+            V1DeploymentList currentDeployments = await this.client.ListDeployments(deviceSelector);
```

Comment:
A rule of thumb when writing code is to reduce scope of living for variable. Other words should always have the smallest amount of lines from variable creation til usage:
```csharp
V1DeploymentList currentDeployments = await this.client.ListDeployments(deviceSelector);
Assert.Single(currentDeployments.Items, d => d.Metadata.Name == moduleName);
V1ServiceAccountList currentServiceAccounts = await this.client.ListServiceAccounts(deviceSelector);
Assert.Single(currentServiceAccounts.Items, sa => sa.Metadata.Name == moduleName);
V1ServiceList currentServices = await this.client.ListServices(deviceSelector);
Assert.Empty(currentServices.Items);
V1PersistentVolumeClaimList currentPvcList = await this.client.ListPeristentVolumeClaims();
Assert.Empty(currentPvcList.Items);
```

Reply:
can be changed:
```csharp
AssertNoDeploymentsExist(deviceSelector, moduleName);
AssertNoServiceAccountsExist(deviceSelector, moduleName);
AssertNoServicesExist(deviceSelector, moduleName);
AssertNoPvcsExist(deviceSelector);
```